rust-runtime/aws-smithy-types/additional-ci

```diff
@@ -5,8 +5,10 @@
 #
 
 # This script contains additional CI checks to run for this specific package
-
 set -e
 
 echo "### Checking for duplicate dependency versions in the normal dependency graph with all features enabled"
 cargo tree -d --edges normal --all-features
+
+echo "### Checking whether the features are properly feature-gated"
+! cargo tree -e no-dev | grep serde
```

Comment:
`!` changes the process exit code.

`cargo tree -e no-dev | grep serde` is expected to fail because we don't want `serde` to be printed when feature gate is not enabled.
grep doesn't find any `serde` in any of the line, so it exits with General Error `1`.

So with `!` the error is going to be considered a success.

Reply:
I hope I'm making sense.

Reply:
```bash

$ false 
$ echo $?
1
$ ! false
$ echo $?
0
$ ! true | false
$ echo $?
0
```
Seems to work. TIL